pkg/server/plugin/datastore/sql/migration.go

```diff
@@ -92,6 +100,16 @@ func initDB(db *gorm.DB) (err error) {
 	return nil
 }
 
+func tableOptionsForDialect(tx *gorm.DB, dbType string) *gorm.DB {
+	// This allows for setting table options for a particular DB type.
+	// For MySQL, (for compatibility reasons) we want to make sure that
+	// we can support indexes on strings (varchar(255) in the DB).
+	if dbType == "mysql" {
+		return tx.Set("gorm:table_options", "ENGINE=InnoDB  ROW_FORMAT=DYNAMIC DEFAULT CHARSET=utf8")
```

Comment:
Random drive-by minor comment: seems like the charset named "utf8mb4" would always be a better choice in mysql than the charset named "utf8", since in mysql "utf8mb4" means "standard utf8", and "utf8" means "a character set invented by the mysql team that's a subset of utf8"

Reply:
Oh, I see that you had a comment about the index error. Ignore me. This compromise probably won't hurt anyone :)